crates/kitsune_p2p/kitsune_p2p/src/spawn/actor.rs

```diff
@@ -184,15 +209,16 @@ impl KitsuneP2pActor {
                                         .await
                                     {
                                         let reason = format!("{:?}", err);
-                                        let fail =
-                                            wire::Wire::failure(reason).encode_vec().unwrap();
-                                        KitsuneMetrics::count(KitsuneMetrics::Fail, fail.len());
-                                        let _ = write.write_and_close(fail).await;
+                                        let fail = wire::Wire::failure(reason);
+                                        let _ = respond
+                                            .respond(fail, KitsuneTimeout::from_millis(1000 * 30))
```

Comment:
Does this mean this `.await` could wait up to 30 seconds? We only have 10 concurrent so this could easily become a bottleneck. Also maybe it should be a const somewhere?

Reply:
Same question for all the responds below.

Reply:
Good call :+1: There was actually already a tuning param for "read" timeouts - I renamed that to "implicit" timeouts so it can work for responds also which are processed in the background.